From 8a9c1d0915e41fdd99b16a5db711f7fd4c922b71 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Fri, 11 Feb 2022 15:47:34 +0100
Subject: [PATCH 484/562] Revert "rtw89: configure rx_filter according to
 FIF_PROBE_REQ"

This reverts commit c0553ccc8e8cdae215a1dda18a9bd5885724b934.
---
 drivers/net/wireless/realtek/rtw89/mac80211.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/drivers/net/wireless/realtek/rtw89/mac80211.c b/drivers/net/wireless/realtek/rtw89/mac80211.c
index 3e4a47642459..00b034999cfc 100644
--- a/drivers/net/wireless/realtek/rtw89/mac80211.c
+++ b/drivers/net/wireless/realtek/rtw89/mac80211.c
@@ -161,7 +161,7 @@ static void rtw89_ops_configure_filter(struct ieee80211_hw *hw,
 	rtw89_leave_ps_mode(rtwdev);
 
 	*new_flags &= FIF_ALLMULTI | FIF_OTHER_BSS | FIF_FCSFAIL |
-		      FIF_BCN_PRBRESP_PROMISC | FIF_PROBE_REQ;
+		      FIF_BCN_PRBRESP_PROMISC;
 
 	if (changed_flags & FIF_ALLMULTI) {
 		if (*new_flags & FIF_ALLMULTI)
@@ -192,15 +192,6 @@ static void rtw89_ops_configure_filter(struct ieee80211_hw *hw,
 			rtwdev->hal.rx_fltr |= B_AX_A_A1_MATCH;
 		}
 	}
-	if (changed_flags & FIF_PROBE_REQ) {
-		if (*new_flags & FIF_PROBE_REQ) {
-			rtwdev->hal.rx_fltr &= ~B_AX_A_BC_CAM_MATCH;
-			rtwdev->hal.rx_fltr &= ~B_AX_A_UC_CAM_MATCH;
-		} else {
-			rtwdev->hal.rx_fltr |= B_AX_A_BC_CAM_MATCH;
-			rtwdev->hal.rx_fltr |= B_AX_A_UC_CAM_MATCH;
-		}
-	}
 
 	rtw89_write32_mask(rtwdev,
 			   rtw89_mac_reg_by_idx(R_AX_RX_FLTR_OPT, RTW89_MAC_0),
-- 
2.34.1

